versioncontrol/github.go

```diff
@@ -0,0 +1,111 @@
+package versioncontrol
+
+import (
+	"dep-report/models"
+	"encoding/json"
+	"fmt"
+	"github.com/pkg/errors"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strings"
+)
+
+func ReportObjFromGithub(dep models.Dependency, r Client) (*models.ReportObject,error) {
+	repoName, err := repoNameFromGithubPackage(dep.Name)
+	if err != nil {
+		return nil, err
+	}
+
+	//consider whether or not we need this var
+	repoURL := "https://api.github.com/repos/" + repoName
+
+	reportObject := models.ReportObject{
+		Name: dep.Name,
+		Website: repoURL,
+		Source: dep.Source,
+	}
+
+	licenseURL := repoURL + "/license"
+	var licenseResponse models.LicenseResponse
+	if err := r.getGithub(licenseURL, &licenseResponse); err != nil {
+		return nil, errors.Wrapf(err, "Unable to get from %s :", licenseURL)
+	}
+
+	reportObject.License = licenseResponse.License.Name
+
+	commitURL := repoURL + "/commits/" + dep.Revision
+	var installed models.CommitResponse
+	if err := r.getGithub(commitURL, &installed); err != nil {
+		return nil, errors.Wrapf(err, "Unable to get from %s :", commitURL)
+	}
+
+	reportObject.Installed = models.VersionDetails{
+		Commit: installed.SHA,
+		Time:   installed.Commit.Committer.Date,
+	}
+
+	branchURL := repoURL + "/commits/HEAD"
+	var latest models.CommitResponse
+	if err := r.getGithub(branchURL, &latest); err != nil {
+		return nil, errors.Wrapf(err, "Unable to get from %s :", branchURL)
+	}
+
+	reportObject.Latest = models.VersionDetails{
+		Commit: latest.SHA,
+		Time:   latest.Commit.Committer.Date,
+	}
+
+	releaseURL := repoURL + "/releases/latest"
+	var release models.Release
+	if err := r.getGithub(releaseURL, &release); err != nil {
+		return nil, errors.Wrapf(err, "Unable to get from %s :", releaseURL)
+	}
+	reportObject.Latest.Version = release.Name
+
+	return &reportObject, nil
+}
+
+func repoNameFromGithubPackage(packageName string) (string, error) {
+	if rawURL, found := GithubRepoURLForPackage[packageName]; found {
+		packageName = rawURL
+	} else {
+		packageName = "https://" + packageName
+	}
+
+	u, err := url.Parse(packageName)
+	if err != nil {
+		return "", fmt.Errorf("unable to parse repo url, %w",err)
+	}
+
+	return strings.Replace(u.Path, "/", "", 1), nil
+}
+
+func (r *Client) getGithub(url string, target interface{}) error {
```

Comment:
I'm of the opinion that object methods should be in the same file as the defined struct, so the file would go:

client.go
```
type Client struct {
...
}

func (c *Client) Foo() error {
...
}

func (c *Client) Bar() error {
...
}

func fooHelper(thing fooable) (fooed){
...
}
```

In my opinion that would greatly increase clarity.

Reply:
If that creates one massive file, then it probably can be broken down further. 

However if that is the case, I would file that as an issue and work on it in the future.

Reply:
Will file issue on this one. Thanks!